RecoEgamma/EgammaPhotonProducers/src/PhotonProducer.cc

```diff
@@ -210,6 +213,17 @@ void PhotonProducer::produce(edm::Event& theEvent, const edm::EventSetup& theEve
   }
   if( validEcalRecHits) endcapRecHits = *(endcapHitHandle.product());
 
+  bool validPreshowerRecHits=true;
+  Handle<EcalRecHitCollection> preshowerHitHandle;
+  theEvent.getByToken(preshowerHits_, preshowerHitHandle);
+  EcalRecHitCollection preshowerRecHits;
+  if (!preshowerHitHandle.isValid()) {
+    edm::LogError("PhotonProducer")
+      << "Error! Can't get the preshowerEcalHits";
+    validPreshowerRecHits=false;
+  }
+  if( validPreshowerRecHits ) preshowerRecHits = *(preshowerHitHandle.product());
```

Comment:
this update reveals multiple atrocities already made in this module: all inputs from the event record are copied to local temporaries just to solve some programming problems.

This has to stop.

Since multiple references were already made to GEDPhotonProducer, please sync up with the choices made there for the product inputs.